### ISSUE TITLE
Fix for CA1822 introduces violation of CA1000

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5320]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: 6.0.100-preview.6.21355.2

### Describe the bug

 [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static` introduces violation of [CA1000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1000): `Do not declare static members on generic types`

### Steps To Reproduce

```csharp
    internal sealed class ReadOnlyBag<T> : IEnumerable
    {
        public bool IsReadOnly => true; // CA1822
```

### Expected behavior

No violation

### Actual behavior

```csharp
    internal sealed class ReadOnlyBag<T> : IEnumerable
    {
        public static bool IsReadOnly => true; // CA1000
```

**Comment by Evangelink** [comment]
Hi @xtqqczze,

Thanks for the feedback! That's an interesting case.

I am wondering if we should just not report or add some user option to control this case because that's still a valuable finding IMO.

For example, in your case, assuming that's as simple, you could do the following:

```csharp
internal sealed class ReadOnlyBag : IEnumerable
{
    public static bool IsReadOnly => true; // No CA1000
}

internal sealed class ReadOnlyBag<T> : ReadOnlyBag
{
```

See this PR #5296 proposal (got rejected because the "problem" is limited to max 2 instances in the case of an analyzer so simplicity was preferred).
